### PR TITLE
Handling of win32 msvc builds.

### PR DESCRIPTION
**Pull request by Twinside**
I got some problem while compiling with MSVC build, due to the ".lib" extension of generated library. Adding a patch to switch between `.a` & `.lib` depending on the platform.

### CODE REVIEW

File: pkg/topkg.ml [hunk 1]
```diff
@@ -27,8 +27,12 @@ module type Exts = sig
   val interface_opt : string list
   (** [interface_opt] is [".cmx" :: interface] *)
 
+  val platform_lib_extension : string
+  (** Extension of the native platform library, ".a" for
```
Comment (dbuenzli) [inline pkg/topkg.ml]: \o/ a Windows user ! 

`Exts.platform_lib_extension` is a very long winded and redundant name. Besides, values from this module are lists of extensions I would suggest: 
```
val c_library : string list 
```
Other than that I'm sure similar business needs to be done for dlls (e.g. [here](https://github.com/dbuenzli/mtime/blob/master/pkg/build.ml#L23)) could you update the PR with a right: 
```
val c_dll_library : string list
```

File: pkg/topkg.ml [hunk 2]
```diff
@@ -144,7 +148,12 @@ end
 module Exts : Exts = struct
   let interface = [".mli"; ".cmi"; ".cmti"]
   let interface_opt = ".cmx" :: interface
-  let library = [".cma"; ".cmxa"; ".cmxs"; ".a"]
+
+  let platform_lib_extension =
```
Comment (dbuenzli) [inline pkg/topkg.ml]: With the new name this should fit in the 80 columns of one line. If it does  please do so and stick it to the other defs.

File: pkg/topkg.ml [hunk 3]
```diff
@@ -229,7 +238,9 @@ module Pkg : Pkg = struct
   let lib =
     let drop_exts =
       if Env.native && not Env.native_dynlink then [ ".cmxs" ] else
-      if not Env.native then [ ".a"; ".cmx"; ".cmxa"; ".cmxs" ] else []
+      if not Env.native then
+        [Exts.platform_lib_extension; ".cmx"; ".cmxa"; ".cmxs" ]
```
Comment (dbuenzli) [inline pkg/topkg.ml]: Same comment as above, at least for the list.

**Comment by Twinside** [comment]
I've added a new commit with your remarks taken in consideration.

**Comment by dbuenzli** [comment]
Perfect ! Could you please squash the two commits together.

**Comment by Twinside** [comment]
And done!

**Comment by dbuenzli** [comment]
Thanks ! Your patch is in as 605fc11a48da220fcef7e0. Contributions to improve Windows support are always *very much* appreciated.

**Pull request closed without merging**